addon/index.js

```diff
@@ -1,28 +1,43 @@
-import Ember from 'ember';
-
-export default function(app, prefix) {
-  var regex = new RegExp('^' + prefix + '\/((?:instance-)?initializers)\/');
-  var getKeys = (Object.keys || Ember.keys);
-  var moduleNames = getKeys(requirejs._eak_seen);
+function resolveInitializer(moduleName) {
+  var module = require(moduleName, null, null, true);
+  if (!module) {
+    throw new Error(moduleName + ' must export an initializer.');
+  }
+  var initializer = module['default'];
+  if (!initializer.name) {
+    initializer.name = moduleName.slice(moduleName.lastIndexOf('/') + 1);
+  }
+  return initializer;
+}
 
+function registerInitializers(app, moduleNames) {
   for (var i = 0; i < moduleNames.length; i++) {
-    var moduleName = moduleNames[i];
-    var matches = regex.exec(moduleName);
-
-    if (matches && matches.length === 2) {
-      var module = require(moduleName, null, null, true);
-      if (!module) { throw new Error(moduleName + ' must export an initializer.'); }
+    app.initializer(resolveInitializer(moduleNames[i]));
+  }
+}
 
-      var initializerType = Ember.String.camelize(matches[1].substring(0, matches[1].length - 1));
-      var initializer = module['default'];
-      if (!initializer.name) {
-        var initializerName = moduleName.match(/[^\/]+\/?$/)[0];
-        initializer.name = initializerName;
-      }
+function registerInstanceInitializers(app, moduleNames) {
+  for (var i = 0; i < moduleNames.length; i++) {
+    app.instanceInitializer(resolveInitializer(moduleNames[i]));
+  }
+}
 
-      if (app[initializerType]) {
-        app[initializerType](initializer);
-      }
+export default function (app, prefix) {
+  var initializerPrefix =  prefix + '/initializers/';
+  var instanceInitializerPrefix =  prefix + '/instance-initializers/';
+  var initializers = [];
+  var instanceInitializers = [];
+  // this is 2 pass because generally the first pass is the problem
+  // and is reduced, and resolveInitializer has potential to deopt
+  var moduleNames = Object.keys(requirejs._eak_seen);
```

Comment:
I'd rather do a major bump and drop IE8 support. 1.13 was a very long time ago, and there are no known issues with the current version of ember-load-initializers that would prevent someone from locking to the current version.